### PR TITLE
Migrate Drag and Drop guide

### LINKED ISSUE #3457: content: migrate HTML5rocks Drag and Drop
* [draft](https://docs.google.com/document/d/1O3-8OkytzyRVAeJ6cd6hlHNNcg5TjygNzdfMrR1Q0X8/edit)
* next step: @rachelandrew to submit PR

history

* @kaycebasques Google Docs draft LGTM
* @kaycebasques reviewed on 7/22

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes #3457

Changes proposed in this pull request:

- Adds updated content based on https://html5rocks.com/en/tutorials/dnd/basics/

### CODE REVIEW

**Comment by rachelandrew** [comment]
I have a video that needs adding to the cloud storage:

[webdev-dnd.zip](https://github.com/GoogleChrome/web.dev/files/4994892/webdev-dnd.zip)

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+title: Using the HTML5 Drag and Drop API
+authors:
+  - ericbidelman
+  - rachelandrew
+date: 2010-09-30
+updated: 2020-07-29
+description:
+  The HTML5 Drag and Drop (DnD) API means that we can make almost any element on our page draggable. In this post I’ll explain the basics of Drag and Drop.
+tags:
+  - html
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:11]:
```suggestion
tags:
  - blog
  - html
  - javascript
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+---
+
+The HTML5 Drag and Drop (DnD) API means that we can make almost any element on our page draggable.
+In this post I'll explain the basics of Drag and Drop.
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:15]:
```suggestion
In this post we'll explain the basics of Drag and Drop.
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
+---
+title: Using the HTML5 Drag and Drop API
+authors:
+  - ericbidelman
+  - rachelandrew
+date: 2010-09-30
+updated: 2020-07-29
+description:
+  The HTML5 Drag and Drop (DnD) API means that we can make almost any element on our page draggable. In this post I’ll explain the basics of Drag and Drop.
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:9]:
```suggestion
description: >
  The HTML5 Drag and Drop (DnD) API means that we can make
  almost any element on our page draggable. In this post we’ll explain
  the basics of Drag and Drop.
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+tags:
+  - html
+---
+
+The HTML5 Drag and Drop (DnD) API means that we can make almost any element on our page draggable.
+In this post I'll explain the basics of Drag and Drop.
+
+## Creating draggable content
+
+It's worth noting that in most browsers, text selections, img elements, and anchor elements are draggable by default.
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:19]:
```suggestion
It's worth noting that in most browsers, text selections, images, and links are draggable by default.
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+If you drag the logo on google.com you will see the ghost image,
+the image can then be dropped in the address bar, an `<input type="file" />` element, or even the desktop.
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:21]:
```suggestion
For example, if you drag the Google logo on [Google Search](https://google.com) you will see the ghost image.
The image can then be dropped in the address bar, an `<input type="file" />` element, or even the desktop.
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+To make other types of content draggable you need to use the HTML5 DnD APIs.
+
+To make an object draggable set `draggable=true` on that element.
+Just about anything can be drag-enabled, images, files, links, files, or any markup on your page.
+
+In my example I am creating an interface to rearrange some columns,
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:27]:
```suggestion
In our example we're creating an interface to rearrange some columns,
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+
+The `dataTransfer` property is where all the DnD magic happens.
+It holds the piece of data sent in a drag action.
+`dataTransfer` is set in the `dragstart` event and read/handled in the drop event.
+Calling `e.dataTransfer.setData(mimeType, dataPayload)` lets you set the object's MIME type and data payload.
+
+In this example, we're going to allow users to rearrange the order of the columns.
+To do that, first you need to store the source element's HTML when the drag starts:
+
+TO DO add video here.
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:234]:
```suggestion
  <figure class="w-figure">
    <video controls autoplay loop muted class="w-screenshot">
      <source src="https://storage.googleapis.com/web-dev-assets/drag-and-drop/webdev-dnd.mp4" type="video/mp4">
    </video>
  </figure>
```

File: src/site/content/en/blog/drag-and-drop/index.md [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+date: 2010-09-30
+updated: 2020-07-29
+description: >
+  The HTML5 Drag and Drop (DnD) API means that we can make
+  almost any element on our page draggable. In this post we’ll explain
+  the basics of Drag and Drop.
+tags:
+  - blog
+  - html
+  - javascript
```
Comment (kaycebasques) [inline src/site/content/en/blog/drag-and-drop/index.md:15]:
```suggestion
  - javascript
  - file-system
```

**Pull request merged**